### PR TITLE
User story 34

### LINKED ISSUE #35: User Story 34
Image

As an admin
When I visit an admin invoice show page (/admin/invoices/:invoice_id)
Then I see all of the items on the invoice including:
- Item name
- The quantity of the item ordered
- The price the Item sold for
- The Invoice Item status

### PR DESCRIPTION
**Pull request by westonio**
# Describe the changes below:
- Adds tests, view, and css for listing the items in a table on the Admin's Invoices Show page 
# Relevant user story:
```
34. Admin Invoice Show Page: Invoice Item Information
As an admin
When I visit an admin invoice show page (/admin/invoices/:invoice_id)
Then I see all of the items on the invoice including:
- Item name
- The quantity of the item ordered
- The price the Item sold for
- The Invoice Item status
```

# Before submitting, check the following:
- [x] Entire test suite is passing
- [100%] SimpleCov test percentage

### CODE REVIEW

File: app/assets/stylesheets/application.css [hunk 1]
```diff
@@ -41,7 +41,7 @@ nav {
 nav ul {
   list-style: none;
   display: flex;
-  padding: 10px 10px 0px 10px
+  padding: 10px 10px 0px 10px;
```
Comment (dani-wilson) [inline app/assets/stylesheets/application.css:44]: those pesky semicolons

File: spec/features/admin/invoices/show_spec.rb [hunk 2]
```diff
@@ -23,14 +23,14 @@
...
 
-      InvoiceItem.create!(item_id: @item_1.id, invoice_id: @invoice_1.id, quantity: 5, unit_price: 70000, status: 0)
-      InvoiceItem.create!(item_id: @item_2.id, invoice_id: @invoice_1.id, quantity: 6, unit_price: 69000, status: 0)
-      InvoiceItem.create!(item_id: @item_3.id, invoice_id: @invoice_2.id, quantity: 5, unit_price: 32000, status: 0)
-      InvoiceItem.create!(item_id: @item_4.id, invoice_id: @invoice_3.id, quantity: 5, unit_price: 16000, status: 0)
-      InvoiceItem.create!(item_id: @item_5.id, invoice_id: @invoice_3.id, quantity: 5, unit_price: 19000, status: 0)
-      InvoiceItem.create!(item_id: @item_5.id, invoice_id: @invoice_4.id, quantity: 7, unit_price: 30000, status: 0)
-      InvoiceItem.create!(item_id: @item_6.id, invoice_id: @invoice_4.id, quantity: 3, unit_price: 20000, status: 0)
-      InvoiceItem.create!(item_id: @item_7.id, invoice_id: @invoice_4.id, quantity: 7, unit_price: 45000, status: 0)
+      @invoice_item_1 = InvoiceItem.create!(item_id: @item_1.id, invoice_id: @invoice_1.id, quantity: 5, unit_price: 70000, status: 0)
```
Comment (MWoodshop) [inline spec/features/admin/invoices/show_spec.rb:26]: Nice use of variable names here for enhanced testing.

File: spec/features/admin/invoices/show_spec.rb [hunk 3]
```diff
@@ -23,14 +23,14 @@
...
+      @invoice_item_2 = InvoiceItem.create!(item_id: @item_2.id, invoice_id: @invoice_1.id, quantity: 6, unit_price: 69000, status: 1)
+      @invoice_item_3 = InvoiceItem.create!(item_id: @item_3.id, invoice_id: @invoice_2.id, quantity: 5, unit_price: 32000, status: 2)
```
Comment (dani-wilson) [inline spec/features/admin/invoices/show_spec.rb:28]: Nice use of assigning variables :D

File: spec/features/admin/invoices/show_spec.rb [hunk 4]
```diff
@@ -51,8 +51,55 @@
       end
 
       it "shows the invoice's Customer first and last name" do
-        expect(page).to have_content(@invoice_1.customer_name, count: 1)
-        expect(page).to_not have_content(@invoice_3.customer.first_name)
+        within("#invoice-customer-info") do
```
Comment (dani-wilson) [inline spec/features/admin/invoices/show_spec.rb:54]: yay within blocks!

File: spec/features/admin/invoices/show_spec.rb [hunk 5]
```diff
@@ -51,8 +51,55 @@
...
+          expect(page).to have_content(@invoice_item_2.item.name, count: 1)
+        end
+
+        expect(page).to_not have_content(@invoice_item_3.item.name)
+        expect(page).to_not have_css("#invoice-item-#{@invoice_item_3.id}")
+      end
+      
+      it "shows each item's quantity ordered" do
+        within("#invoice-item-#{@invoice_item_1.id}") do
+          expect(page).to have_content(@invoice_item_1.quantity)
```
Comment (dani-wilson) [inline spec/features/admin/invoices/show_spec.rb:77]: tests look really nice!

**Review by dani-wilson — APPROVED** [review]
Awesome work! Everything looks good!

**Review by MWoodshop — APPROVED** [review]
This all looks well thought out and correct to me. On a later branch simplecov has app/controllers/admin/invoices_controller.rb as a missed line so we might need to revisit that if this or the other pending PRs doesn't cover it.

**Pull request merged**